Sources/StreamChatUI/CommonViews/AvatarView/ChatChannelAvatarView.swift

```diff
@@ -30,31 +47,293 @@ open class ChatChannelAvatarView: _View, ThemeProvider, SwiftUIRepresentable {
                 preferredSize: .avatarThumbnailSize,
                 components: components
             )
+            presenceAvatarView.isOnlineIndicatorVisible = false
             return
         }
 
-        let (avatarURL, isOnlineIndicatorVisible): (URL?, Bool) = {
-            // Try to get the explicit channel avatar first
-            if let avatarURL = channel.imageURL {
-                return (avatarURL, false)
-            }
-
-            // TODO: https://stream-io.atlassian.net/browse/CIS-652
-            // this is just placeholder implementation:
-
-            let firstOtherMember = channel.lastActiveMembers
-                .sorted { $0.memberCreatedAt < $1.memberCreatedAt }
-                .first(where: { $0.id != content.currentUserId })
-
-            return (firstOtherMember?.imageURL, firstOtherMember?.isOnline ?? false)
-        }()
-
+        loadAvatar(for: channel)
+    }
+    
+    open func loadAvatar(for channel: ChatChannel) {
+        // If the channel has an avatar set, load that avatar
+        if let channelAvatarUrl = channel.imageURL {
+            loadChannelAvatar(from: channelAvatarUrl)
+            return
+        }
+      
+        // Use the appropriate method to load avatar based on channel type
+        if channel.isDirectMessageChannel {
+            loadDirectMessageChannelAvatar(channel: channel)
+        } else {
+            loadMergedAvatars(channel: channel)
+        }
+    }
+    
+    /// Loads the avatar from the URL. This function is used when the channel has a non-nil `imageURL`
+    /// - Parameter url: The `imageURL` of the channel
+    open func loadChannelAvatar(from url: URL) {
         presenceAvatarView.avatarView.imageView.loadImage(
-            from: avatarURL,
+            from: url,
             placeholder: appearance.images.userAvatarPlaceholder4,
             preferredSize: .avatarThumbnailSize,
             components: components
         )
+    }
+    
+    /// Loads avatar for a directMessageChannel
+    /// - Parameter channel: The channel
+    open func loadDirectMessageChannelAvatar(channel: ChatChannel) {
+        let lastActiveMembers = channel.lastActiveMembers
+            .sorted { $0.memberCreatedAt < $1.memberCreatedAt }
```

Comment:
In case when we will have 100500 participants this one can impact on execution speed. 
The complexity of sorting is big, means time required will also be big enough (O(n log n)). 
I see couple of options here:
1. make all counting in background
2. add field to data entity with `lastActiveMemebers` with array of 3-4 participants, but this will possible require backend and internal DB changes

@tbarbugli can you advice here please?